### PR TITLE
chore: add prerelease cors

### PR DESCRIPTION
**Pull request by eidam**

### CODE REVIEW

File: packages/prerelease-registry/functions/_middleware.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const onRequestOptions: PagesFunction = async () => {
+  return new Response(null, {
+    headers: {
+      "Access-Control-Allow-Origin": "*",
+      "Access-Control-Allow-Methods": "GET,OPTIONS",
+      "Access-Control-Max-Age": "86400",
+    },
+  });
+};
```
Comment (GregBrimble) [inline packages/prerelease-registry/functions/_middleware.ts:9]: This is another way of doing it that should mean you don't need the change in `getArtifactForWorkflowRun`.

```suggestion
export const onRequest: PagesFunction = async ({ next }) => {
  const response = await next()
  response.headers.set("Access-Control-Allow-Origin", "*");
  response.headers.set("Access-Control-Allow-Methods", "GET,OPTIONS");
  response.headers.set("Access-Control-Max-Age", "86400");
  return response;
};
```

Comment (GregBrimble) [reply packages/prerelease-registry/functions/_middleware.ts:9]: Maybe, IDK. Maybe an `OPTIONS` call would be weird. If that doesn't work, you could have an `onRequestOptions` and `onRequest` in this file: the first returns what you had. The second returns what I suggested. Then that'll cover all endpoints.

**Review by GregBrimble — APPROVED** [review]
But also, what you have looks like it'll work too anyway so whatever :)

File: packages/prerelease-registry/functions/_middleware.ts (hunk 1 above)
Comment (eidam) [reply packages/prerelease-registry/functions/_middleware.ts:9]: oh nice! yeah, I think we need to keep options returning the response, and then just adding headers to everything else

**Comment by threepointone** [comment]
Let's land this and see how it goes!

**Pull request merged**